perfkitbenchmarker/publisher.py

```diff
@@ -227,12 +228,16 @@ def _FormatMetadata(self, metadata):
                     for k, v in sorted(metadata.iteritems()))
 
   def PublishSamples(self, samples):
+    stream = io.BytesIO()
     dashes = '-' * 25
-    self.stream.write('\n' + dashes +
-                      'PerfKitBenchmarker Results Summary' +
-                      dashes + '\n')
+    stream.write('\n' + dashes +
```

Comment:
This change is in order to print the results to two locations: `self.stream` and the log.  The local variable `stream` is used to build the full formatted text output, then that message (via `stream.getvalue()`) is written to `self.stream` and `logging.debug`.

Would a different variable name make this clearer? Perhaps `buffer`?